sycl/include/CL/sycl/handler.hpp

```diff
@@ -1607,11 +1607,7 @@ class __SYCL_EXPORT handler {
     size_t OneElemSize =
         IsTreeReduction ? sizeof(typename Reduction::result_type) : 0;
     uint32_t NumConcurrentWorkGroups =
-#ifdef __SYCL_REDUCTION_NUM_CONCURRENT_WORKGROUPS
```

Comment:
Why did you remove this macro read here?
Moving it to reduction.cpp is not the same as having it here.
The idea of this macro was to let user (and SYCL developers) experiment with it during compilation of the test/bench.
clang++ -fsycl my_bench.cpp -D__SYCL_REDUCTION_NUM_CONCURRENT_WORKGROUPS=16

Moving it to reduction.cpp means that variable would be ignored during my_bench.cpp compilation. Changing that var would require rebuild of libsycl.so